### PR TITLE
Assure Maven 3.1.0 compatibility and fix dependencies

### PR DESCRIPTION
**Pull request by fvgh**
#391 reported an incompatibility of the spotless-maven-plugin with Maven 3.3.9.
The [Spotless Maven ReadMe](https://github.com/diffplug/spotless/blob/master/plugin-maven/README.md) claims that the spotless-maven-plugin is compatible with Maven 3.1.0.
Research suggested that the problem is related to version conflicts regarding provided dependencies with other plugins. A [similar issue](https://github.com/spotbugs/spotbugs-maven-plugin/issues/21) occurred in the Spotbugs Maven plugin.

Due to [Maven dependency mediation](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Transitive_Dependencies), it seems incorrect to use newer dependencies than required and provided by Maven 3.1.0.

The original problem reported in #391 could not be reproduced.
But problems with Maven 3.1.0 showed that the [Maven System class loader](https://maven.apache.org/guides/mini/guide-maven-classloading.html) is a transitive dependency of the Spotless plugin, which should not be the case. Further investigation showed that this was most likely not caused by spotless.

Current PR just improves Maven plugin test and dependencies.

### CODE REVIEW

**Comment by nedtwigg** [comment]
Since a full year has elapsed, I'm guessing this will end up not getting merged.  Happy to reopen or discuss further if you'd like to revisit.

**Comment by nedtwigg** [comment]
I don't know anything about maven. If you'd like to merge this PR @fvgh feel free, but probably best to first request a review from lutovich in case they've got any opinions.

**Comment by fvgh** [comment]
@lutovich I am afraid I am not a Maven expert myself. I started this PR, since the initial #391 description suggested some spotless-eclipse-base problems with SLF4J and the underlying Maven. But this was not the case. The problem I see with the Plexus Classworld should not only affect Maven 3.1.0, but also higher version. Can you have a look? It is not urgent, since the issue is anyhow open since many month.

File: plugin-maven/build.gradle [hunk 1]
```diff
@@ -69,7 +69,9 @@ dependencies {
 		implementation "com.diffplug.spotless:spotless-lib-extra:${libVersion}"
 	}
 
-	implementation "org.codehaus.plexus:plexus-resources:${VER_PLEXUS_RESOURCES}"
+	implementation("org.codehaus.plexus:plexus-resources:${VER_PLEXUS_RESOURCES}") {
+		exclude group: 'classworld', module: 'classworld'
```
Comment (lutovich) [inline plugin-maven/build.gradle:73]: Is it "classworld" or "classworld**s**"?
I think the chain of transitive dependencies goes like this:
```
org.codehaus.plexus:plexus-resources:jar:1.0.1:compile
+- org.codehaus.plexus:plexus-utils:jar:3.0.8:compile
\- org.codehaus.plexus:plexus-container-default:jar:1.0-alpha-9-stable-1:compile
   \- classworlds:classworlds:jar:1.1-alpha-2:compile
```

File: gradle.properties [hunk 2]
```diff
@@ -29,6 +29,6 @@ VER_MOCKITO=3.3.3
 
 # Used for Maven Plugin
 VER_MAVEN_API=3.0
-VER_ECLIPSE_AETHER=1.1.0
+VER_ECLIPSE_AETHER=0.9.0.M2
```
Comment (lutovich) [inline gradle.properties:32]: Makes sense to use the same Aether version as the minimum Maven version that's supported by the plugin 👍

**Review by lutovich — COMMENTED** [review]
@fvgh thank you for investigating this problem! It does look very complex. I added one comment on this PR.

**Comment by lutovich** [comment]
Would it be okay to bump `plexus-resources` from `1.0.1` to `1.2.0`? The latter does not depend on `plexus-container-default` and does not have `classworlds` as a transitive dependency.

File: plugin-maven/build.gradle (hunk 1 above)
Comment (fvgh) [reply plugin-maven/build.gradle:73]: Thanks for the heads-up. I need another look...

**Comment by fvgh** [comment]
> Would it be okay to bump `plexus-resources` from `1.0.1` to `1.2.0`? The latter does not depend on `plexus-container-default` and does not have `classworlds` as a transitive dependency.

This was also an idea. But I found the reasoning behind the change in [plexus-resources #2](https://github.com/codehaus-plexus/plexus-resources/issues/2) unrelated to the issue I found. So I just want to avoid regression for now. Anyhow, I have to have another look about the reproducibility of the error and the assumed solution.

File: plugin-maven/build.gradle (hunk 1 above)
Comment (fvgh) [reply plugin-maven/build.gradle:73]: I am afraid the problem I have found during investigation of #391 is sporadic, hence I assumed that my findings fixed the problem. So to conclude:
* Problem described in #391 cannot be reproduced.
* Problem found in during investigation of #391 is sporadic.
* I have no time to investigate any further. Will reword this PR and make it an enhancement.

**Comment by fvgh** [comment]
@lutovich Please have a final look. All the changes are just enhancements, but not bug fixes.

**Review by lutovich — APPROVED** [review]

**Pull request merged**